### DISCUSSION TITLE
How to hide, then restore a window ?

### DISCUSSION THREAD

**Discussion by kivimango** [slint-ui/slint#4362] [Q&A]
Hello !
I would like to ask to how can i hide the window with the possibility to restore it later ?
My use case is: there is  a splash screen, then a child process is spawned with the Command::new() function.
When the child process is exited, i need to restore the previous window to its initial state.
When i call window.hide(), then window.show(), the window wont open.
I guess because the handler is droppen when i call hide() according to the [docs](https://slint.dev/releases/1.3.0/docs/rust/slint/struct.Window#method.hide):

_Hides the window, so that it is not visible anymore. The additional strong reference on the associated component, that was created when [Self::show()](https://slint.dev/releases/1.3.0/docs/rust/slint/struct.Window.html#method.show) was called, is dropped._

I tried to change the width and height of the window to 0, then set it again, but its a hack, and sometimes just don't work.
When i want to create a new window, the application panics.
The window has to be full-screen.

**Comment by hunger** [comment]
`hide()` and `show()` are the right approach. By default the event loop gets shut down when the last window closes, which usually quits the application.

The `master` branch has `slint::run_event_loop_until_quit()` for that.
